### PR TITLE
Bug1536693 5.7

### PR DESCRIPTION
**Pull request by laurynas-biveinis**
http://jenkins.percona.com/job/mysql-5.7-param/159/

### CODE REVIEW

File: storage/innobase/include/os0event.h [hunk 1]
```diff
@@ -27,8 +27,219 @@ Created 2012-09-23 Sunny Bains (split from os0sync.h)
...
+	typedef CONDITION_VARIABLE	os_cond_t;
+#else
+/** Native condition variable */
+	typedef pthread_cond_t		os_cond_t;
+#endif /* _WIN32 */
+
+	os_cond_t		cond_var;	/*!< condition variable is
+						used in waiting for the event */
+
+protected:
```
Comment (percona-ysorokin) [inline storage/innobase/include/os0event.h]: private ?

File: storage/innobase/include/os0event.h [hunk 2]
```diff
@@ -27,8 +27,219 @@ Created 2012-09-23 Sunny Bains (split from os0sync.h)
...
+	// Disable copy constructor
+	os_event(const os_event&);
```
Comment (percona-ysorokin) [inline storage/innobase/include/os0event.h:237]: hide assignment operator as well

File: storage/innobase/include/sync0rw.h [hunk 3]
```diff
@@ -682,20 +682,24 @@ struct rw_lock_t
 #ifdef UNIV_DEBUG
 /** Value of rw_lock_t::magic_n */
 # define RW_LOCK_MAGIC_N	22643
+#endif /* UNIV_DEBUG */
 
 	/** Constructor */
 	rw_lock_t()
 	{
-		magic_n = RW_LOCK_MAGIC_N;
```
Comment (percona-ysorokin) [inline storage/innobase/include/sync0rw.h]: Are the following two lines equivalent in Release mode

File: storage/innobase/include/sync0rw.h [hunk 4]
```diff
@@ -682,20 +682,24 @@ struct rw_lock_t
...
+		ut_d(magic_n = RW_LOCK_MAGIC_N;)
```
Comment (percona-ysorokin) [inline storage/innobase/include/sync0rw.h:690]: What about getting rid of "magic_n" class member unless UNIV_DEBUG is defined?

**Comment by percona-ysorokin** [comment]
LGTM

File: storage/innobase/include/os0event.h (hunk 2 above)
Comment (laurynas-biveinis) [inline storage/innobase/include/os0event.h:237]: rwlocks get assigned in the R-tree code, and with os_events inlined into rwlocks, the assignment operator is now needed too. By code review it appeared that the compiler-generated one is suitable

File: storage/innobase/include/os0event.h (hunk 1 above)
Comment (laurynas-biveinis) [inline storage/innobase/include/os0event.h]: Was copied over from os0event.cc without changes. No problem to change it now, same maintenance cost

File: storage/innobase/include/sync0rw.h (hunk 4 above)
Comment (laurynas-biveinis) [inline storage/innobase/include/sync0rw.h:690]: It is already so, isn't it?

Comment (percona-ysorokin) [inline storage/innobase/include/sync0rw.h:690]: My bad, got confused with collapsed code

**Comment by laurynas-biveinis** [comment]
Review comments addressed and UNIV_UNLIKELY changes carried over from the 5.6 version. @percona-ysorokin, please review again.
http://jenkins.percona.com/job/mysql-5.7-param/164/

**Comment by percona-ysorokin** [comment]
Looks good

**Pull request merged**